libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -924,4 +928,106 @@ export class TeamsActivityHandler extends ActivityHandler {
             await handler(teamsChannelData.team, context, next);
         });
     }
+
+    /**
+     * Runs the _event_ sub-type handlers, as appropriate, and then continues the event emission process.
+     *
+     * @param context The context object for the current turn.
+     * @returns A promise that represents the work queued.
+     *
+     * @remarks
+     * Overwrite this method to support channel-specific behavior across multiple channels or to add
+     * custom event sub-type events.
+     */
+    protected async dispatchEventActivity(context: TurnContext): Promise<void> {
+        await this.handle(context, 'Event', async () => {
+            if (context.activity.channelId == Channels.Msteams) {
+                switch (context.activity.name) {
+                    case 'application/vnd.microsoft.meetingStart':
+                        await this.onTeamsMeetingStart(context);
+                        break;
+                    case 'application/vnd.microsoft.meetingEnd':
+                        await this.onTeamsMeetingEnd(context);
+                        break;
```

Comment:
Note: I originally had these `return await...`, but switched just now to simply `await` so as not to shortcut the parent, `ActivityHandler.dispatchEventActivity()`. However, I'm not sure it would ever *need* to *also* handle the event, so I'm open to reverting that commit.

Reply:
Edit: Reverted. On second thought, the other "event" handlers all `return await` and it doesn't make sense to continue handling these events longer than necessary.